### PR TITLE
Pi-hole core v3.2

### PR DESCRIPTION
**Pull request by DL6ER**
Release

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -1434,99 +1995,139 @@ main() {
...
     install_dependent_packages DEPS[@]
 
+    if [[ -x "$(command -v systemctl)" ]]; then
+      # Value will either be 1, if true, or 0
+      LIGHTTPD_ENABLED=$(systemctl is-enabled lighttpd | grep -c 'enabled' || true)
+    else
+      # Value will either be 1, if true, or 0
+      LIGHTTPD_ENABLED=$(service lighttpd status | awk '/Loaded:/ {print $0}' | grep -c 'enabled' || true)
+    fi
+
```
Comment (bcambl) [inline automated install/basic-install.sh]: Lines 2037 - 2043 is duplicate of 2009 - 2015.

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -1434,99 +1995,139 @@ main() {
...
     fi
     install_dependent_packages DEPS[@]
 
+    if [[ -x "$(command -v systemctl)" ]]; then
+      # Value will either be 1, if true, or 0
+      LIGHTTPD_ENABLED=$(systemctl is-enabled lighttpd | grep -c 'enabled' || true)
+    else
+      # Value will either be 1, if true, or 0
+      LIGHTTPD_ENABLED=$(service lighttpd status | awk '/Loaded:/ {print $0}' | grep -c 'enabled' || true)
+    fi
```
Comment (bcambl) [inline automated install/basic-install.sh]: On a clean install: 
`LIGHTTPD_ENABLED` defaults to `0` due to the service being disabled and stopped by default.
This results in `lighttpd` not being enabled and started upon a clean install.

File: automated install/basic-install.sh [hunk 3]
```diff
@@ -1434,99 +1995,139 @@ main() {
...
 
-  if [[ ${INSTALL_WEB} == true ]]; then
-    start_service lighttpd
-    enable_service lighttpd
-  fi
+  # If the Web server was installed,
+  if [[ "${INSTALL_WEB}" == true ]]; then
 
-  runGravity
+    if [[ "${LIGHTTPD_ENABLED}" == "1" ]]; then
```
Comment (bcambl) [inline automated install/basic-install.sh]: Defaults to `0` as `lighttpd` is stopped and disabled by default. (ref line 2011)

File: automated install/basic-install.sh [hunk 4]
```diff
@@ -1195,174 +1653,273 @@ ${additional}" ${r} ${c}
...
     fi
     binary="pihole-FTL-linux-x86_32"
   fi
 
-  FTLinstall "${binary}" || return 1
+  #In the next section we check to see if FTL is already installed (in case of pihole -r).
+  #If the installed version matches the latest version, then check the installed sha1sum of the binary vs the remote sha1sum. If they do not match, then download
+  echo -e "  ${INFO} Checking for existing FTL binary..."
+
+  local ftlLoc=$(which pihole-FTL)
```
Comment (bcambl) [inline automated install/basic-install.sh]: We should hide the following stdout:
```
  [i] Checking for existing FTL binary...
which: no pihole-FTL in (/usr/local/sbin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin)
  [✓] Downloading and Installing FTL

```

File: automated install/basic-install.sh [hunk 5]
```diff
@@ -855,195 +1228,244 @@ install_dependent_packages() {
...
-    install -D ${PI_HOLE_LOCAL_REPO}/advanced/{index,blockingpage}.* /var/www/html/pihole/
-    if [ -f /var/www/html/index.lighttpd.html ]; then
-      mv /var/www/html/index.lighttpd.html /var/www/html/index.lighttpd.orig
-    else
-      printf "\n:::\tNo default index.lighttpd.html file found... not backing up"
-    fi
-    echo " done!"
+    # don't do anything
+    echo -e "${OVER}  ${CROSS} ${str}
+    No default index.lighttpd.html file found... not backing up"
```
Comment (bcambl) [inline automated install/basic-install.sh]: Indent 2 spaces:
```
  [✓] Creating directory for blocking page, and copying files
  [✗] Backing up index.lighttpd.html
    No default index.lighttpd.html file found... not backing up
```

File: automated install/basic-install.sh (hunk 1 above)
Comment (PromoFaux) [reply automated install/basic-install.sh]: Duplicate on purpose as it's in two different branches of an it's in the `else` branch of `if [[ "${userUpdateVars}" == false ]]`.

**Pull request merged**